tools/walletextension/api/routes.go

```diff
@@ -301,10 +301,18 @@ func authenticateRequestHandler(walletExt *walletextension.WalletExtension, conn
 		return
 	}
 
-	// get message from the request
-	message, ok := reqJSONMap[common.JSONKeyMessage]
-	if !ok || message == "" {
-		handleError(conn, walletExt.Logger(), fmt.Errorf("unable to read message field from the request"))
+	// get address from the request
+	address, ok := reqJSONMap[common.JSONKeyAddress]
+	if !ok || address == "" {
+		handleError(conn, walletExt.Logger(), fmt.Errorf("unable to read address field from the request"))
+		return
+	}
+
+	// TODO: Do we need to have userID if we can get it from query params?
```

Comment:
I don't follow what is going on here?

Reply:
Here is a quick example to demonstrate it (from Gateway frontend):

```
 const authenticateUserURL = pathAuthenticate+"?u="+userID
  const authenticateFields = {"signature": signature, "userID": userID, "address": account }
  const authenticateResp = await fetch(
      authenticateUserURL, {
        method: methodPost,
        headers: jsonHeaders,
        body: JSON.stringify(authenticateFields)
      }
  );
  ```
  
  As you can see we are sending `userID` in `authenticateFields` and we are sending that request to `authenticateUserURL` which also contains `userID`. Currently we parse both (as in design document) and check if they match.
  Potential improvement is not to send `userID` in `authenticateFields` and only get it from URL. What do you think?
  It is a very minor thing.

Reply:
what if we remove it from the query params?